f5-bigip-mixin/alerts/alerts.libsonnet

```diff
@@ -0,0 +1,83 @@
+{
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'f5-bigip-alerts',
+        rules: [
+          {
+            alert: 'F5BigIPLowNodeAvailabilityStatus',
+            expr: |||
+              100 * (sum(bigip_node_status_availability_state) / clamp_min(count(bigip_node_status_availability_state), 1)) < %(alertsCriticalNodeAvailability)s
+            ||| % $._config,
+            'for': '5m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              summary: 'Detecting a significant number of unavailable nodes which can causes potential downtime or degraded performance.',
+              description:
+                (
+                  '{{ printf "%%.0f" $value }} percent of available nodes, ' +
+                  'which is below the threshold of %(alertsCriticalNodeAvailability)s.'
+                ) % $._config,
+            },
+          },
+          {
+            alert: 'F5BigIPServerSideConnectionLimit',
+            expr: |||
+              max without(instance, job) (100 * bigip_node_serverside_cur_conns / clamp_min(bigip_node_serverside_max_conns, 1)) > %(alertsWarningServerSideConnectionLimit)s
+            ||| % $._config,
+            'for': '5m',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              summary: 'Approaching the connection limit may lead to rejecting new connections, impacting availability.',
+              description:
+                (
+                  '{{ printf "%%.0f" $value }} percent of the max number of connections in use on node {{$labels.node}}, ' +
+                  'which is above the threshold of %(alertsWarningServerSideConnectionLimit)s percent.'
+                ) % $._config,
+            },
+          },
+          {
+            alert: 'F5BigIPHighRequestRate',
+            expr: |||
+              max without(instance, job) (100 * increase(bigip_pool_tot_requests[10m]) / clamp_min(increase(bigip_pool_tot_requests[1h]), 1)) > %(alertsCriticalHighRequestRate)s
+            ||| % $._config,
+            'for': '10m',
+            labels: {
+              severity: 'critical',
```

Comment:
should this be critical? it seems like this could happen normally maybe?  like if there's not much traffic on a site. and then there is a surge because of the time of day...would this trigger?

Reply:
yeah, if there is hardly any traffic and then a check-in starting time for a particular application, then the threshold would have a surge suddenly. I can switch this to warning. (This is with the offset or a rate thought process)